### PR TITLE
bpo-44491: Allow clearing the sqlite3 authoriser callback

### PR DESCRIPTION
**Pull request by erlend-aasland**
https://bugs.python.org/issue44491

### CODE REVIEW

File: Doc/library/sqlite3.rst [hunk 1]
```diff
@@ -430,6 +430,8 @@ Connection Objects
       argument and the meaning of the second and third argument depending on the first
       one. All necessary constants are available in the :mod:`sqlite3` module.
 
+      Passing :const:`None` as *authorizer_callback* will disable the authorizer.
```
Comment (pablogsal) [inline Doc/library/sqlite3.rst:433]: Please, add a `versionadded` or `versionchanged`

File: Modules/_sqlite/connection.c [hunk 2]
```diff
@@ -1052,20 +1052,24 @@ pysqlite_connection_set_authorizer_impl(pysqlite_Connection *self,
...
+    int rc;
+    if (authorizer_cb == Py_None) {
+        rc = sqlite3_set_authorizer(self->db, NULL, NULL);
+        Py_XSETREF(self->function_pinboard_authorizer_cb, NULL);
+    }
+    else {
+        Py_INCREF(authorizer_cb);
+        Py_XSETREF(self->function_pinboard_authorizer_cb, authorizer_cb);
+        rc = sqlite3_set_authorizer(self->db, _authorizer_callback, authorizer_cb);
+    }
```
Comment (pablogsal) [inline Modules/_sqlite/connection.c:1069]: ~Aren't you pupulating `self->function_pinboard_authorizer_cb` even if `sqlite3_set_authorizer` fails?~

Ah, I see you are unsetting it later in `        Py_XSETREF(self->function_pinboard_authorizer_cb, NULL);`

File: Doc/library/sqlite3.rst (hunk 1 above)
Comment (erlend-aasland) [reply Doc/library/sqlite3.rst:433]: Ah dang, of course. Thanks.

File: Modules/_sqlite/connection.c [hunk 3]
```diff
@@ -1052,20 +1052,24 @@ pysqlite_connection_set_authorizer_impl(pysqlite_Connection *self,
...
-
     if (!pysqlite_check_thread(self) || !pysqlite_check_connection(self)) {
         return NULL;
     }
 
-    rc = sqlite3_set_authorizer(self->db, _authorizer_callback, (void*)authorizer_cb);
+    int rc;
+    if (authorizer_cb == Py_None) {
+        rc = sqlite3_set_authorizer(self->db, NULL, NULL);
+        Py_XSETREF(self->function_pinboard_authorizer_cb, NULL);
```
Comment (pablogsal) [inline Modules/_sqlite/connection.c:1062]: This should be just:

```suggestion
        Py_XDECREF(self->function_pinboard_authorizer_cb);
```

**Comment by erlend-aasland** [comment]
Syncing with `main` to avoid spurious ssl failures in CI :)

**Review by pablogsal — APPROVED** [review]

**Pull request merged**